lib/datasets/mnist.rb

```diff
@@ -0,0 +1,106 @@
+require 'zlib'
+
+require_relative "dataset"
+
+module Datasets
+  class MNIST < Dataset
+    module Pixelable
+      def pixels
+        data.unpack("C*")
+      end
+
+      def to_h
+        hash = super
+        hash[:pixels] = pixels
+        hash
+      end
+    end
+
+    class Record < Struct.new(:data, :label)
+      include Pixelable
+    end
+
+    def initialize(type: :train)
+      unless [:train, :test].include?(type)
+        raise 'Please set type :train or :test'
```

Comment:
`raise ArgumentError, "type must be :train or :test: #{type.inspect}` is better.
It uses suitable exception class and has the actual value.

Reply:
(We should improve CIFAR dataset later.)

Reply:
Thanks for your teaching! I did not know `.inspect` method, It's really useful!
I fixed.